providers/dns/googledomains/googledomains.go

```diff
@@ -0,0 +1,133 @@
+package googledomains
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-acme/lego/v4/challenge"
+	"github.com/go-acme/lego/v4/challenge/dns01"
+	"github.com/go-acme/lego/v4/platform/config/env"
+	"google.golang.org/api/acmedns/v1"
+	"google.golang.org/api/option"
+)
+
+// Environment variables names.
+const (
+	envNamespace = "GOOGLE_DOMAINS_"
+
+	EnvAccessToken = envNamespace + "ACCESS_TOKEN"
+	EnvZoneName    = envNamespace + "ZONE_NAME"
+
+	EnvPropagationTimeout = envNamespace + "PROPAGATION_TIMEOUT"
+	EnvPollingInterval    = envNamespace + "POLLING_INTERVAL"
+	EnvHTTPTimeout        = envNamespace + "HTTP_TIMEOUT"
+)
+
+// static compile-time check on interface implementation.
+var _ challenge.Provider = &DNSProvider{}
+
+// Config is used to configure the creation of the DNSProvider.
+type Config struct {
+	AccessToken        string
+	PollingInterval    time.Duration
+	PropagationTimeout time.Duration
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider.
+func NewDefaultConfig() *Config {
+	return &Config{
+		AccessToken:        env.GetOrDefaultString(EnvAccessToken, ""),
+		PropagationTimeout: env.GetOrDefaultSecond(EnvPropagationTimeout, 2*time.Minute),
+		PollingInterval:    env.GetOrDefaultSecond(EnvPollingInterval, 2*time.Second),
+	}
+}
+
+// NewDNSProvider returns the Google Domains DNS provider with a default configuration.
+func NewDNSProvider() (*DNSProvider, error) {
+	return NewDNSProviderConfig(NewDefaultConfig())
+}
```

Comment:
```suggestion
func NewDNSProvider() (*DNSProvider, error) {
	values, err := env.Get(EnvAccessToken)
	if err != nil {
		return nil, fmt.Errorf("googledomains: %w", err)
	}

	config := NewDefaultConfig()
	config.AccessToken = values[EnvAccessToken]

	return NewDNSProviderConfig(config)
}
```

Reply:
I'm confused at this suggestion, don't we do the same thing with NewDefaultConfig()?

Reply:
`NewDefaultConfig` must not be dependent on env vars.
`NewDNSProvider` is only for env vars.

Reply:
Ah I see, I think I saw most other providers having NewDefaultConfig use env.GetOrDefaultString, should the NewDefaultConfig be written differently?